File: src/main/java/com/GoldenMine/object/Point2D.java
package com.GoldenMine.object;

import com.GoldenMine.parser.Variable;
import org.jetbrains.annotations.NotNull;

public class Point2D implements KoreanObject {
    double x;
    double y;

    // a와 b로 좌표를 만듭니다.
    // new Point2D(x, y);
    // 10과 30으로 xy좌표를 만듭니다.

    @Override
    public String getType() {
        return "좌표";
    }

    @Override
    public Variable.VariableMode getValueType(String key) {
        switch(key) {
            case "x":
            case "X":
                return Variable.VariableMode.REALNUM_MODE;
        }

        return Variable.VariableMode.NULL;
    }

    @Override
    public Object getValue(String key) {
        return null;
    }

    @Override
    public void setValue(String key, Object value) {

    }

    @Override
    public void setValue(int index, Object value) {

    }

    @Override
    public int compareTo(@NotNull KoreanObject o) {
        return 0;
    }
}

File: src/main/java/kr/goldenmine/object/Point2D.kt
package kr.goldenmine.`object`

class Point2D : KoreanObject {
    internal var x: Double = 0.toDouble()
    internal var y: Double = 0.toDouble()

    // a와 b로 좌표를 만듭니다.
    // new Point2D(x, y);
    // 10과 30으로 xy좌표를 만듭니다.

    override fun getValue(key: String): Any? {
        when (key) {
            "x", "X" -> return x
            "y", "Y" -> return y
        }
        return null
    }

    override fun setValue(key: String, value: Any?) {
        when (key) {
            "x", "X" -> this.x = value as Double
            "y", "Y" -> this.y = value as Double
        }
    }

    override fun setValue(index: Int, value: Any?) {
        when (index) {
            0 -> this.x = value as Double
            1 -> this.y = value as Double
        }
    }

    override fun compareTo(o: KoreanObject): Int {
        return if (o is Point2D) {

            if (x < o.x) {
                -1
            } else if (x > o.x) {
                1
            } else if (y < o.y) {
                -1
            } else if (y > o.y) {
                1
            } else {
                0
            }
        } else {
            0
        }
    }
}
